packages/mask/src/plugins/Trader/trader/useAllTradeComputed.ts

```diff
@@ -187,6 +187,19 @@ export function useAllTradeComputed(
     )
     const openoceanSwapEstimateGas = useDODOTradeGasLimit(openocean as TradeComputed<SwapRouteData> | null)
 
+    const isTraderJoe = tradeProviders.some((x) => x === TradeProvider.TRADERJOE)
+    // traderjoe
+    const traderjoe_ = useUniswapV2Trade(
+        TradeProvider.TRADERJOE,
+        TradeStrategy.ExactIn,
+        inputAmount_,
+        '0',
+        isTraderJoe ? inputToken : undefined,
+        isTraderJoe ? outputToken : undefined,
+    )
+    const traderjoe = useUniswapTradeComputed(traderjoe_.value, inputToken, outputToken)
+    const traderjoeEstimateGas = useUniswapTradeGasLimit(traderjoe, TradeProvider.TRADERJOE)
```

Comment:
extract as hook function?

Reply:
@septs I have updated the code, do you want this?